libs/src/sdk/services/DiscoveryNodeSelector/DiscoveryNodeSelector.ts

```diff
@@ -27,6 +28,18 @@ import EventEmitter from 'events'
 import { AbortController as AbortControllerPolyfill } from 'node-abort-controller'
 import { mergeConfigWithDefaults } from '../../utils/mergeConfigs'
 
+const getPathFromUrl = (url: string) => {
```

Comment:
will look into using url instead of regex, was worried about cross platform compat

Reply:
this works for both paths and full urls tho :D. but won't work for non http/https protocols lol